### PR TITLE
8295270: RISC-V: Clean up and refactoring for assembler functions

### PR DESCRIPTION
**Pull request by RealFYang**
Witnessed that some high-level assember functions are placed in file assembler_riscv.hpp/cpp,
such as 'movptr', 'li' and so on. These are macro-assembler functions which should be placed
in macroAssembler_riscv.hpp/cpp. Meanwhile, we should also move load & store memory and
control assembler functions from address of type 'address' / 'Address'.

Testing: Tier1 on HiFive Unmatched board {fastdebug, release}.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8295270](https://bugs.openjdk.org/browse/JDK-8295270): RISC-V: Clean up and refactoring for assembler functions

### Reviewers
 * [Feilong Jiang](https://openjdk.org/census#fjiang) (@feilongjiang - Author) ⚠️ Review applies to [d98ab458](https://git.openjdk.org/jdk/pull/10697/files/d98ab45895124b6e8670701ba12596e4022c2606)
 * [Yadong Wang](https://openjdk.org/census#yadongwang) (@yadongw - Author) ⚠️ Review applies to [d98ab458](https://git.openjdk.org/jdk/pull/10697/files/d98ab45895124b6e8670701ba12596e4022c2606)
 * [Aleksey Shipilev](https://openjdk.org/census#shade) (@shipilev - **Reviewer**) ⚠️ Review applies to [25e1baa2](https://git.openjdk.org/jdk/pull/10697/files/25e1baa2dda4a5b0afe3970bf719fd2449267ca6)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10697/head:pull/10697` \
`$ git checkout pull/10697`

Update a local copy of the PR: \
`$ git checkout pull/10697` \
`$ git pull https://git.openjdk.org/jdk pull/10697/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10697`

View PR using the GUI difftool: \
`$ git pr show -t 10697`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10697.diff">https://git.openjdk.org/jdk/pull/10697.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/cpu/riscv/macroAssembler_riscv.hpp [hunk 1]
```diff
@@ -562,6 +660,215 @@ class MacroAssembler: public Assembler {
...
+
+#define INSN(NAME)                                                                                 \
+  void NAME(Register Rd, address dest) {                                                           \
+    assert_cond(dest != NULL);                                                                     \
+    int64_t offset = dest - pc();                                                                  \
+    if (is_offset_in_range(offset, 32)) {                                                          \
+      auipc(Rd, (int32_t)offset + 0x800);                                                          \
+      Assembler::NAME(Rd, Rd, ((int32_t)offset << 20) >> 20);                                      \
+    } else {                                                                                       \
+      int32_t offset = 0;                                                                          \
```
Comment (feilongjiang) [inline src/hotspot/cpu/riscv/macroAssembler_riscv.hpp:678]: It might be confusing that we have already defined `offset` in L673 with a different type.

**Review by feilongjiang — APPROVED** [review]
Change looks good. Thanks.

**Review by yadongw — APPROVED** [review]
lgtm

**Comment by RealFYang** [comment]
@feilongjiang @feilongjiang : Thanks for reviewing this non-trivial change. Need a Reviewer then. @shipilev : Want to take a look? Thanks.

File: src/hotspot/cpu/riscv/assembler_riscv.hpp [hunk 2]
```diff
@@ -2719,6 +2412,10 @@ enum Nf {
 #undef FUNC
 
 public:
+bool do_compress() const {
+  return UseRVC && in_compressible_region();
+}
```
Comment (shipilev) [inline src/hotspot/cpu/riscv/assembler_riscv.hpp:2417]: Indenting had been lost.

File: src/hotspot/cpu/riscv/macroAssembler_riscv.hpp [hunk 3]
```diff
@@ -449,28 +447,83 @@ class MacroAssembler: public Assembler {
...
   void bgez(Register Rs, Label &l, bool is_far = false);
   void bltz(Register Rs, Label &l, bool is_far = false);
   void bgtz(Register Rs, Label &l, bool is_far = false);
+
+  void beq(Register Rs1, Register Rs2, Label &L, bool is_far = false);
+  void bne(Register Rs1, Register Rs2, Label &L, bool is_far = false);
+  void blt(Register Rs1, Register Rs2, Label &L, bool is_far = false);
+  void bge(Register Rs1, Register Rs2, Label &L, bool is_far = false);
+  void bltu(Register Rs1, Register Rs2, Label &L, bool is_far = false);
+  void bgeu(Register Rs1, Register Rs2, Label &L, bool is_far = false);
```
Comment (shipilev) [inline src/hotspot/cpu/riscv/macroAssembler_riscv.hpp:480]: Do you want to line things like these up? For example:

```

  void beq (Register Rs1, Register Rs2, Label &L, bool is_far = false);
  void bne (Register Rs1, Register Rs2, Label &L, bool is_far = false);
  void blt (Register Rs1, Register Rs2, Label &L, bool is_far = false);
  void bge (Register Rs1, Register Rs2, Label &L, bool is_far = false);
  void bltu(Register Rs1, Register Rs2, Label &L, bool is_far = false);
  void bgeu(Register Rs1, Register Rs2, Label &L, bool is_far = false);
```

**Review by shipilev — APPROVED** [review]
Looks okay from a brief look.

File: src/hotspot/cpu/riscv/macroAssembler_riscv.hpp (hunk 3 above)
Comment (RealFYang) [reply src/hotspot/cpu/riscv/macroAssembler_riscv.hpp]: Yes, that looks better. Fixed.

**Pull request closed without merging**